tensorflow_datasets/core/lazy_imports_lib_test.py

```diff
@@ -58,6 +60,27 @@ def test_bad_import(self):
     with self.assertRaisesWithPredicateMatch(ImportError, "extras_require"):
       _ = tfds.core.lazy_imports.test_foo
 
+  # pylint: disable=import-outside-toplevel
+  def test_lazy_import_context_manager(self):
```

Comment:
Thank you. To make sure the lazy importer isn't globally added, could you also add a test to make sure importing outside the lazy importer still works ?
```py
import nltk
self.assertTrue(hasattr(nltk, 'function'))
```
And that unknown import still fails, even when added to `_ALLOWED_LAZY_DEPS`.
```py
_ALLOWED_LAZY_DEPS.append('valid_module')
self.assertRaisesRegex(ImportError, '^Import error 'valid_module'$'):
  import valid_module
```